src/sandstorm/supervisor.capnp

```diff
@@ -274,15 +256,28 @@ struct ApiTokenOwner {
 
       introducerIdentity @9 :Text;
       # The identity ID through which a user's powerbox action caused the grain to receive this
-      # token. This is the identity against which the `requiredPermissions` parameter
-      # to `restore()` will be checked. This field is only intended to be filled in by the
-      # front-end during a powerbox request; a regular `save()` call produces a capability that
-      # has no "introducer".
+      # token. The token will become invalid if this identity loses access to the grain. This field
```

Comment:
FWIW, I personally tend to favor migrations where possible over needing to remember to support both structures forever.  It turns an ongoing maintenance cost/conceptual debt (how should the code behave when it encounter tokens that have this shape + educating/reminding devs that once upon a time we didn't enforce the invariants that new tokens uphold) into one that can be paid down all-at-once.

I think it's higher cost to teach future devs (including my future self, who will likely have forgotten) about various inconsistencies than it is to remove them from existence.

That said, I'm also wary of asking other people to write possibly tricky code, and it seems the other implementation already lives, so I'm reasonably happy with either of David's suggestions, and if I want to put my money where my mouth is, I could write a migration for this at ~any time in the future.